codec-http/src/test/java/io/netty/handler/codec/http/HttpRequestEncoderTest.java

```diff
@@ -48,6 +48,16 @@ public void testUriWithoutPath2() throws Exception {
     }
 
     @Test
+    public void testUriWithEmptyPath() throws Exception {
+        HttpRequestEncoder encoder = new HttpRequestEncoder();
+        ByteBuf buffer = Unpooled.buffer(64);
+        encoder.encodeInitialLine(buffer, new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET,
+                "http://localhost:9999/?p1=v1"));
+        String req = buffer.toString(Charset.forName("US-ASCII"));
+        assertEquals("GET http://localhost:9999/?p1=v1 HTTP/1.1\r\n", req);
```

Comment:
Looks like we should add `buffer.release();` here and for all tests in this class.